crates/egui/src/widgets/button.rs

```diff
@@ -243,188 +246,62 @@ impl Widget for Button<'_> {
             button_padding.y = 0.0;
         }
 
-        let (space_available_for_image, right_text_font_height) = if let Some(text) = &text {
-            let font_height = ui.fonts(|fonts| text.font_height(fonts, ui.style()));
-            (
-                Vec2::splat(font_height), // Reasonable?
-                font_height,
-            )
-        } else {
-            (
-                ui.available_size() - 2.0 * button_padding,
-                default_font_height(),
-            )
-        };
+        let mut prepared = layout
+            .frame(Frame::new().inner_margin(button_padding))
+            .allocate(ui);
+
+        let response = if ui.is_rect_visible(prepared.response.rect) {
+            let visuals = ui.style().interact_selectable(&prepared.response, selected);
+
+            if image_tint_follows_text_color {
+                prepared.sized_atomics.iter_mut().for_each(|a| {
+                    a.kind = match std::mem::take(&mut a.kind) {
+                        SizedAtomicKind::Image(image, size) => {
+                            SizedAtomicKind::Image(image.tint(visuals.text_color()), size)
+                        }
+                        other => other,
```

Comment:
Maybe it could be nice to have helpers for this? Something like

```suggestion
                prepared.sized_atomics.for_each_image_mut(|image, _size| {
                     *image = image.tint(visuals.text_color());
```